File: 0796-rotate-string/0796-rotate-string.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
        if(s.size()!=goal.size()) return false;
        string g = goal+goal;
        int n=s.size();
        for(int i=0;i<g.size();i++){
            string sub = g.substr(i,n);
            if(sub==s) return true;
        }
        return false;
    }
};

File: 0796-rotate-string/0796-rotate-string.java
class Solution {
    public boolean rotateString(String s, String goal) {
        if(s.length()!=goal.length()) return false;

        //step 1: concate
        goal = goal+goal;
        
        //step 2: use 'contains' method of java (in C++, we can use 'find' method)
        return goal.contains(s);
    }
}